src/golang/lib/lambda/utils.go

```diff
@@ -314,3 +323,59 @@ func AddFunctionTypeToChannel(functionsToShip []LambdaFunctionType, channel chan
 		channel <- lambdaFunctionTypeToPass
 	}
 }
+
+func DeleteAllDockerImages(functionsToShip []LambdaFunctionType) error {
+	for _, lambdaFunctionType := range functionsToShip {
+		lambdaImageUri, _, err := mapFunctionType(lambdaFunctionType)
+		if err != nil {
+			return errors.Wrap(err, "Unable to map function type to image.")
+		}
+		versionedLambdaImageUri := fmt.Sprintf("%s:%s", lambdaImageUri, lib.ServerVersionNumber)
+		err = DeleteDockerImage(versionedLambdaImageUri)
+		if err != nil {
+			return errors.Wrapf(err, "Unable to delete docker image of %s", versionedLambdaImageUri)
+		}
+	}
+	return nil
+}
+
+func DeleteDockerImage(versionedLambdaImageUri string) error {
+	// Remove Docker Image after finish creating Lambda functions.
+
+	stdout := &bytes.Buffer{}
+	stderr := &bytes.Buffer{}
+
+	cmd := exec.Command(
+		"docker",
+		"images",
+		fmt.Sprintf("--filter=reference=%s", versionedLambdaImageUri),
+		"-q")
+
+	cmd.Stdout = stdout
+	cmd.Stderr = stderr
+
+	err := cmd.Run()
+	if err != nil {
+		stdoutString := strings.TrimSpace(stdout.String())
+		stderrString := strings.TrimSpace(stderr.String())
+		return errors.Wrapf(err, "Unable to find the docker image. Stdout: %s. Stderr: %s.", stdoutString, stderrString)
+	}
+
+	imageId := strings.TrimSpace(stdout.String())
+
+	// The docker image does not exist or has been deleted so we can't delete it again.
+	if imageId == "" {
+		return nil
+	}
+
+	cmd = exec.Command("docker", "rmi", "-f", imageId)
+
+	err = cmd.Run()
+	if err != nil {
+		stdoutString := strings.TrimSpace(stdout.String())
+		stderrString := strings.TrimSpace(stderr.String())
+		return errors.Wrapf(err, "Unable to delete docker image. Stdout: %s. Stderr: %s.", stdoutString, stderrString)
```

Comment:
@Fanjia-Yan we haven't cleared the `stdout` and `stderr` buffers and we haven't set `cmd.Stdout = stdout` and `cmd.Stderr = stderr` either, so I don't think they are capturing the right errors now? Have you tested if this logs the error appropriately?

Reply:
Good catch! I have cleared the `stdout` and `stderr` byte buffer using` Reset()` and setup the `cmd` properly again. Have tested it, the error logs match the `exec.Command`.